File: android-tv/app/src/main/java/com/pontoview/telas/BootReceiver.java
package com.pontoview.telas;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null) return;
        String action = intent.getAction();
        if (!Intent.ACTION_BOOT_COMPLETED.equals(action) && !Intent.ACTION_LOCKED_BOOT_COMPLETED.equals(action)) return;

        SharedPreferences prefs = context.getSharedPreferences("pontoview_player", Context.MODE_PRIVATE);
        if (!prefs.getBoolean("auto_start", true)) return;

        try {
            Intent launch = new Intent(context, MainActivity.class);
            launch.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
            context.startActivity(launch);
        } catch (Exception ignored) {
        }
    }
}

File: android-player/app/src/main/java/br/com/pontoview/player/BootReceiver.kt
package br.com.pontoview.player

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class BootReceiver : BroadcastReceiver() {

    companion object {
        private const val TAG = "PontoViewBoot"
    }

    override fun onReceive(context: Context, intent: Intent) {
        val validAction =
            intent.action == Intent.ACTION_BOOT_COMPLETED ||
            intent.action == Intent.ACTION_LOCKED_BOOT_COMPLETED ||
            intent.action == Intent.ACTION_MY_PACKAGE_REPLACED

        if (!validAction || !DeviceConfig.isAutoStartEnabled(context)) return

        launchPlayer(context)

        val pending = goAsync()
        Thread {
            try {
                Thread.sleep(3500)
                launchPlayer(context)
            } catch (error: Exception) {
                Log.w(TAG, "Delayed autostart attempt failed", error)
            } finally {
                pending.finish()
            }
        }.start()
    }

    private fun launchPlayer(context: Context) {
        try {
            val launchIntent = Intent(context, MainActivity::class.java).apply {
                addFlags(
                    Intent.FLAG_ACTIVITY_NEW_TASK or
                        Intent.FLAG_ACTIVITY_CLEAR_TOP or
                        Intent.FLAG_ACTIVITY_SINGLE_TOP
                )
                putExtra("started_from_boot", true)
            }
            context.startActivity(launchIntent)
            Log.i(TAG, "PontoView Player launch requested after boot")
        } catch (error: Exception) {
            Log.w(TAG, "Android blocked PontoView Player autostart", error)
        }
    }
}
